### PR TITLE
Add retry for gpt api call and improve air_bench aggregation function

### PR DESCRIPTION
**Pull request by pbcong**
- Issue: Too Many Requests errors in API calls; so I added retry for each call

- Improve air_bench_foundation aggregation to return result for each tasks rather than the total average

### CODE REVIEW

File: lmms_eval/tasks/clotho_aqa/utils.py [hunk 1]
```diff
@@ -49,9 +49,13 @@ def clotho_aqa_doc_to_text(doc, lmms_eval_specific_kwargs):
...
     }
+
 elif API_TYPE == "azure":
-    API_URL = os.getenv("AZURE_ENDPOINT", "https://api.cognitive.microsoft.com/sts/v1.0/issueToken")
-    API_KEY = os.getenv("AZURE_API_KEY", "YOUR_API_KEY")
+    API_KEY = os.getenv("AZURE_API_KEY", None)
+    deployment_name = os.getenv("AZURE_DEPLOYMENT_NAME", None)
+    resource_name = os.getenv("AZURE_RESOURCE_NAME", None)
+    api_version = os.getenv("AZURE_API_VERSION", None)
+    API_URL = f"https://{resource_name}.openai.azure.com/openai/deployments/{deployment_name}/chat/completions?api-version={api_version}"
```
Comment (kcz358) [inline lmms_eval/tasks/clotho_aqa/utils.py]: A small question on how we format this azure api url. Why don't we simple export the formatted to `API_URL` and then use the original format. This keeps your env clean and less export operation needed.

**Review by kcz358 — COMMENTED** [review]
Just a small question for how you format the AZURE API URL. Can we just simply export the formatted URL in one go and use the original format. I have seen many different URL format for azure api format so I feel it is important to keep the same

**Comment by pbcong** [comment]
Hi! Thank you for pointing that out. I’ve made the changes to the code.

**Review by kcz358 — COMMENTED** [review]
Hi, thank you for the changes. You can set the default endpoint to `"https://api.cognitive.microsoft.com/sts/v1.0/issueToken"` if no endpoint is provided. Just like what we did in llava_wilder and many other tasks

https://github.com/EvolvingLMMs-Lab/lmms-eval/blob/df3cb0f0dc42b865e429467b671917575cefa543/lmms_eval/tasks/llava_wilder/utils.py#L42-L48

**Comment by pbcong** [comment]
I updated, thank you!

**Review by kcz358 — APPROVED** [review]

**Pull request merged**